docs/content/doc/advanced/hacking-on-gitea.en-us.md

```diff
@@ -136,30 +136,36 @@ You should lint, vet and spell-check with:
 make vet lint misspell-check
 ```
 
-### Updating the stylesheets
+### Updating CSS
 
-To generate the stylsheets, you will need [Node.js](https://nodejs.org/) at version 8.0 or above.
+To generate the CSS, you will need [Node.js](https://nodejs.org/) 8.0 or greater and the build dependencies:
 
-At present we use [less](http://lesscss.org/) and [postcss](https://postcss.org) to generate our stylesheets. Do
-**not** edit the files in `public/css/` directly, as they are generated from
-`lessc` from the files in `public/less/`.
+```bash
+npm install
+```
 
-If you wish to work on the stylesheets, you will need to install `lessc` the
-less compiler and `postcss`. The recommended way to do this is using `npm install`:
+At present we use [less](http://lesscss.org/) and [postcss](https://postcss.org) to generate our CSS. Do
```

Comment:
Wouldn't mind if we added mention of what needs to be installed directly on this page in addition to just linking to the pages of those projects.
```bash
npm install less -g
npm install postcss
npm install clean-css
```
(Or whatever the appropriate commands are if not those)

Reply:
`npx` is used, so the user won't need to install anything globally.

Reply:
`npm install` takes care of it all. It installes bin scripts into `node_modules/.bin` which `npx` then calls directly.

Reply:
ugh ok my bad I thought I had run npm install and then didn't. FWIW since I had not, it did fail with this:
```
[mrsdizzie@mbp ~/.go/src/code.gitea.io/gitea (pr-6952-1557933284) ] >  make css
npx lesshint public/less/
npx: installed 178 in 5.333s
npx lessc --clean-css="--s0 -b" public/less/index.less public/css/index.css
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/lessc - Not found
npm ERR! 404
npm ERR! 404  'lessc@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url
```

Reply:
Actually this is a bug. The package `less` provides binary `lessc`, so we should actually use `npx less`. Investigating possible fixes.

Reply:
@silverwind I've just pushed a fix. by running `npx -p less lessc....` things work.

Reply:
@mrsdizzie what version does `npm -v` show for you?

The canonical fix for this naming issue would be `npx -p less lessc` but that argument [has its own bugs](https://github.com/zkat/npx/issues/214). I'm thinking we could avoid this issue entirely by not using `npx` and specifying relative paths to the bin scripts. This would also enable us to remove the dependency on `npx` (which unfortunatly, seems rather unmaintained itself).

Reply:
@techknowlogick I don't consider it a fix because of the linked issue. It needlessly reinstalls the package every execution.

Reply:
```bash
npm -v
6.9.0
```

Reply:
I control the `lessc` package on npm, and so if we need to use that we can.

Reply:
Weird, I see different results with no `lessc` in my path on the same version:

````
npx lesshint public/less/
npx: installed 178 in 5.409s
npx lessc --clean-css="--s0 -b" public/less/index.less public/css/index.css
npx: installed 61 in 3.81s
command not found: lessc
make: *** [css] Error 1
````

Guess I'll change the make scripts to not use npx and abort if `node_modules` is absent, it's the safest route.

Reply:
Converting to `npx` was done to ease Windows developers.   
We used to call the bin scripts via relative path, but Windows had a fit.

Reply:
Oh, right, I forgot. I pushed a fixed with relative paths, but I guess I'll revert.

Reply:
@techknowlogick ok let's go with your version. I hope that eventually npx will fix that reinstallation bug, but it's no blocker.

Reply:
seems `npx -p less lessc --clean-css...` didn't work without `npm i` due to plugin for lessc not being installed.

Reply:
Which plugin is missing? Maybe try multiple `-p` arguments.

Reply:
Specifically `less-plugin-clean-css`. I'm going to pull down the branch and attempt multiple `-p` args.

Reply:
I pushed a fix with multiple `-p`, was also necessary for postcss.

Now because of the mentioned bug with `-p`, modules are installed every execution even if `node_modules` is present. I'm thinking maybe I should just re-add the check for presence of `node_modules` and if its there, assume bin scripts work via regular `npx` calls.

Reply:
Pushed the fix with the check for `node_modules`. I think this is the best compromise as the user only has to install modules once and subsequent runs of `make css` and `make js` will be fast that way and we can be sure that all plugins and stuff will be there as well.